src/HotChocolate/Core/src/Types.Scalars/EmailAddressType.cs

```diff
@@ -0,0 +1,104 @@
+using HotChocolate.Language;
+using HotChocolate.Types.Scalars;
+
+namespace HotChocolate.Types
+{
+    /// <summary>
+    /// A field whose value conforms to the standard internet email address format as specified in HTML Spec
+    /// </summary>
+    public class EmailAddressType : StringType
+    {
+        /// <summary>
+        /// Initializes a new instance of the <see cref="EmailAddressType"/> class.
+        /// </summary>
+
+        public EmailAddressType()
+            : this(
+                WellKnownScalarTypes.EmaillAddress,
+                ScalarResources.EmailAddressType_Description)
+        {
+        }
+
+        /// <summary>
+        /// Initializes a new instance of the <see cref="EmailAddressType"/> class.
+        /// </summary>
+        public EmailAddressType(
+            NameString name,
+            string? description = null,
+            BindingBehavior bind = BindingBehavior.Explicit)
+            : base(name, description, bind)
+        {
+            Description = description;
+        }
+
+        /// <inheritdoc />
+        protected override bool IsInstanceOfType(string runtimeValue)
+        {
+            return runtimeValue != string.Empty;
+        }
+
+        /// <inheritdoc />
+        protected override bool IsInstanceOfType(StringValueNode valueSyntax)
+        {
+            return valueSyntax.Value != string.Empty;
+        }
+
+        /// <inheritdoc />
+        protected override string ParseLiteral(StringValueNode valueSyntax)
+        {
+            var rgx = Regex.matches(valueSyntax, @"/^\+[1-9]\d{1,14}$/");
```

Comment:
Regarding the regex, i suggest to use the following one:

I would also suggest to create the regular expression as a compiled expression, because it is much faster


So the code could look something like this: 
```csharp
    /// <summary>
    /// Well established regex for email validation
    /// Source : https://emailregex.com/
    /// </summary>
    private static readonly string _validationPattern = @"(?:[a-z0-9!#$%&'*+/=?^_`{|}~-]+(?:\.[a-z0-9!#$%&'*+/=?^_`{|}~-]+)*|""(?:[\x01-\x08\x0b\x0c\x0e-\x1f\x21\x23-\x5b\x5d-\x7f]|\\[\x01-\x09\x0b\x0c\x0e-\x7f])*"")@(?:(?:[a-z0-9](?:[a-z0-9-]*[a-z0-9])?\.)+[a-z0-9](?:[a-z0-9-]*[a-z0-9])?|\[(?:(?:25[0-5]|2[0-4][0-9]|[01]?[0-9][0-9]?)\.){3}(?:25[0-5]|2[0-4][0-9]|[01]?[0-9][0-9]?|[a-z0-9-]*[a-z0-9]:(?:[\x01-\x08\x0b\x0c\x0e-\x1f\x21-\x5a\x53-\x7f]|\\[\x01-\x09\x0b\x0c\x0e-\x7f])+)\])";
    
    private static readonly Regex _validationRegex = 
        new Regex(
               _validationPattern,
               RegexOptions.Compiled | RegexOptions.IgnoreCase);
```

Reply:
Yeah, I originally had it as compiled but then I read https://docs.microsoft.com/en-us/dotnet/standard/base-types/best-practices which seems to suggest that for a few functions it's not worth using compiled. But I'll switch it back

Reply:
I guess it makes sense in this case as this will be executed on each Mail field multiple time per request.
So if we can make it faster, lets make it faster :)